client/js/components/procedure/StatementSegmentsList/StatementMeta/StatementMetaLocationAndDocumentReference.vue

```diff
@@ -0,0 +1,365 @@
+        secondary
+        @primary-action="dpValidateAction('statementLocationAndDocumentReference', save, false)"
+        @secondary-action="reset" />
+    </div>
+  </fieldset>
+</template>
+
+<script>
+import {
+  DpButton,
+  DpButtonRow,
+  DpSelect,
+  dpValidateMixin
+} from '@demos-europe/demosplan-ui'
+import { mapActions, mapState } from 'vuex'
+
+export default {
+  name: 'StatementMetaLocationAndDocumentReference',
+
+  components: {
+    DpButton,
+    DpButtonRow,
+    DpMapModal: () => import('@DpJs/components/statement/assessmentTable/DpMapModal'),
+    DpSelect
+  },
+
+  mixins: [dpValidateMixin],
+
+  props: {
+    editable: {
+      type: Boolean,
+      required: false,
+      default: false
+    },
+
+    initiallySelectedDocumentId: {
+      type: String,
+      required: false,
+      default: ''
+    },
+
+    initiallySelectedElementId: {
+      type: String,
+      required: false,
+      default: ''
+    },
+
+    initiallySelectedParagraphId: {
+      type: String,
+      required: false,
+      default: ''
+    },
+
+    procedureId: {
+      type: String,
+      required: true
+    },
+
+    statement: {
+      type: Object,
+      required: true
+    }
+  },
+
+  data () {
+    return {
+      localStatement: null,
+      selectedDocumentId: this.initiallySelectedDocumentId,
+      selectedElementId: this.initiallySelectedElementId,
+      selectedParagraphId: this.initiallySelectedParagraphId
+    }
+  },
+
+  computed: {
+    ...mapState('ElementsDetails', {
+      elements: 'items'
+    }),
+
+    documentOptions () {
+      const documents = this.getDocuments()
+
+      return documents.length > 0
+        ? documents.map(document => ({
+          label: document.attributes.title,
+          value: document.id
+        }))
+        : []
+    },
+
+    elementsOptions () {
+      return Object.values(this.elements).map(element => ({
+        label: element.attributes.title,
+        value: element.id
+      })
+      )
+    },
+
+    isButtonRowDisabled () {
+      const elementIsChanged = this.selectedElementId !== this.initiallySelectedElementId
+      const paragraphIsChanged = this.selectedParagraphId !== this.initiallySelectedParagraphId
+      const documentIsChanged = this.selectedDocumentId !== this.initiallySelectedDocumentId
+
+      return !elementIsChanged
+        || (elementIsChanged && ((this.paragraphOptions.length > 0 && !paragraphIsChanged) || (this.documentOptions.length > 0 && !documentIsChanged)))
+    },
+
+    paragraphOptions () {
+      const paragraphs = this.getParagraphs()
+
+      return paragraphs.length > 0
+        ? paragraphs.map(paragraph => ({
+          label: paragraph.attributes.title,
+          value: paragraph.id
+        }))
+        : []
+    },
+
+    selectedDocumentTitle () {
+      const documents = this.getDocuments()
+
+      return documents.length > 0
+        ? documents.find(document => document.id === this.selectedDocumentId)?.attributes?.title
```

Comment:
In case the find method returns `undefined` is it intended to return `undefined` here or can't this happen?
Same for  `selectedParagraphTitle`

Reply:
Good catch, fixed in 7d1ed500215721a03c3b3bc0183b04a780da3a12